tools/istio-docker.mk

```diff
@@ -18,166 +18,93 @@
+$(ISTIO_OUT)/istio_ca.crt $(ISTIO_OUT)/istio_ca.key: ${GEN_CERT} | ${ISTIO_OUT}
+	${GEN_CERT} --key-size=2048 --out-cert=${ISTIO_OUT}/istio_ca.crt \
+                    --out-priv=${ISTIO_OUT}/istio_ca.key --organization="k8s.cluster.local" \
                     --self-signed=true --ca=true
-$(ISTIO_DOCKER)/node_agent.crt $(ISTIO_DOCKER)/node_agent.key: ${GEN_CERT} $(ISTIO_DOCKER)/istio_ca.crt $(ISTIO_DOCKER)/istio_ca.key
-	${GEN_CERT} --key-size=2048 --out-cert=${ISTIO_DOCKER}/node_agent.crt \
-                    --out-priv=${ISTIO_DOCKER}/node_agent.key --organization="NodeAgent" \
-                    --host="nodeagent.google.com" --signer-cert=${ISTIO_DOCKER}/istio_ca.crt \
-                    --signer-priv=${ISTIO_DOCKER}/istio_ca.key
-
-# directives to copy files to docker scratch directory
-
-# tell make which files are copied form go/out
-DOCKER_FILES_FROM_ISTIO_OUT:=pilot-test-client pilot-test-server pilot-test-eurekamirror \
-                             pilot-discovery pilot-agent sidecar-injector servicegraph mixs \
-                             istio_ca flexvolume node_agent multicluster_ca
-$(foreach FILE,$(DOCKER_FILES_FROM_ISTIO_OUT), \
-        $(eval $(ISTIO_DOCKER)/$(FILE): $(ISTIO_OUT)/$(FILE) | $(ISTIO_DOCKER); cp $$< $$(@D)))
-
-# This generates rules like:
-#$(ISTIO_DOCKER)/pilot-agent: $(ISTIO_OUT)/pilot-agent | $(ISTIO_DOCKER)
-# 	cp $$< $$(@D))
-
-# tell make which files are copied from the source tree
-DOCKER_FILES_FROM_SOURCE:=pilot/docker/prepare_proxy.sh docker/ca-certificates.tgz tools/deb/envoy_bootstrap_tmpl.json \
-                          $(PROXY_JSON_FILES) $(NODE_AGENT_TEST_FILES) $(FLEXVOLUMEDRIVER_FILES) $(GRAFANA_FILES) \
-                          pilot/docker/certs/cert.crt pilot/docker/certs/cert.key
-$(foreach FILE,$(DOCKER_FILES_FROM_SOURCE), \
-        $(eval $(ISTIO_DOCKER)/$(notdir $(FILE)): $(FILE) | $(ISTIO_DOCKER); cp $(FILE) $$(@D)))
-
-# Copy the envoy images, but use standard file names (without the version) in docker
-$(ISTIO_DOCKER)/envoy: ${ISTIO_ENVOY_RELEASE_PATH} | $(ISTIO_DOCKER); cp ${ISTIO_ENVOY_RELEASE_PATH} $(ISTIO_DOCKER)/envoy
-$(ISTIO_DOCKER)/envoy-debug: ${ISTIO_ENVOY_DEBUG_PATH} | $(ISTIO_DOCKER); cp ${ISTIO_ENVOY_DEBUG_PATH} $(ISTIO_DOCKER)/envoy-debug
+$(ISTIO_OUT)/node_agent.crt $(ISTIO_OUT)/node_agent.key: ${GEN_CERT} $(ISTIO_OUT)/istio_ca.crt $(ISTIO_OUT)/istio_ca.key
+	${GEN_CERT} --key-size=2048 --out-cert=${ISTIO_OUT}/node_agent.crt \
+                    --out-priv=${ISTIO_OUT}/node_agent.key --organization="NodeAgent" \
+                    --host="nodeagent.google.com" --signer-cert=${ISTIO_OUT}/istio_ca.crt \
+                    --signer-priv=${ISTIO_OUT}/istio_ca.key
 
 # pilot docker images
 
-docker.eurekamirror: $(ISTIO_DOCKER)/pilot-test-eurekamirror
-docker.proxy_init: $(ISTIO_DOCKER)/prepare_proxy.sh
-docker.sidecar_injector: $(ISTIO_DOCKER)/sidecar-injector
-
-docker.proxy: tools/deb/envoy_bootstrap_tmpl.json ${PROXY_JSON_FILES}
```

Comment:
Please stop deleting my changes - I spent quite a bit of time removing the template to make it easier to understand - we want to use make as a light automation tool that most devs can understand, not turn it into bazel.

Reply:
By "template" are you referring to DOCKER_RULE?  That never got removed and instead one-off custom rules were added for proxy and pilot that seemed to add no value at all to the general population (nor at they described anywhere--e.g., what's the value of  "Dockerfile.appdbg"?)  "docker" seems to have a suitable ARG mechanism that can avoid the need for sed (and ARG is also being leveraged by the other PR for supporting PowerPC).

For using Dockerfile.proxy_debug in place of Dockerfile.proxy, that should just be a matter of running "docker tag" to retag the "proxy_debug" image as "debug".

Reply:
Yes, I removed it and added explicit rules for proxy and pilot, as we had before the rule was added. And I hope in time all other images will follow this model. 

We can use ARG instead of appdbg - but if you are not using it or seeing any value I don't think it's reasonable to delete it, there are far more important things to do and maybe someone else is using it.